### DISCUSSION TITLE
Whisper-powered lyrics extraction from songs 🎵

### DISCUSSION THREAD

**Discussion by eshiofune** [openai/whisper#1363] [Show and tell]
I wrote an article [here](https://medium.com/@ehioroboevans/extracting-song-lyrics-using-the-openai-whisper-model-21eeb19aa722) describing how to make use of either the Whisper API or models to extract song lyrics from audio files.
I was able to get up to 89% similarity with the lyrics found on Google for the Adele song Set Fire to the Rain.

The code is available [here](https://github.com/eshiofune/whisper-lyrics).

**Comment by phineas-pta** [comment]
there're already so many tutorials/articles on how to use whisper, to further expand your article and improve accuracy, you should try out source separation models (Spleeter or Demucs) to extract vocals, VAD (SileroVAD or pyannote-audio) to remove silence / background-music-only segments, creating subtitle for karaoke, etc.

**Reply by eshiofune** [reply]
Thanks for the suggestions !
Creating subtitles for karaoke is right up my alley. Will look into that

**Reply by EtienneAb3d** [reply]
See:
https://github.com/EtienneAb3d/WhisperHallu

**Reply by eshiofune** [reply]
> See: https://github.com/EtienneAb3d/WhisperHallu

@EtienneAb3d Yes I also noticed the hallucination effect around the end of the song and reading the [thread](https://github.com/openai/whisper/discussions/679) you referenced in your repo and of course the repo itself shows some interesting approaches to reduce this effect. Thanks ! I'll take note of this for improving accuracy.
